### PR TITLE
TRUNK-5679  Add tests for Lucene Analyzer Factory

### PR DESCRIPTION
**Pull request by brandones**
https://issues.openmrs.org/browse/TRUNK-5679

Seeking feedback. Not sure if this is the right approach or not, but I'm trying to follow https://dzone.com/articles/quick-intro-hibernate-search

### CODE REVIEW

File: api/src/test/java/org/openmrs/api/db/hibernate/search/LuceneAnalyzerFactoryTest.java [hunk 1]
```diff
@@ -0,0 +1,86 @@
+/**
```
Comment (brandones) [inline api/src/test/java/org/openmrs/api/db/hibernate/search/LuceneAnalyzerFactoryTest.java]: This is the relevant file in this PR; the others are from https://github.com/openmrs/openmrs-core/pull/3040

**Comment by mogoodrich** [comment]
Thanks @brandones ... assumedly there's a way to test existing objects that have been annotated, like PersonName?  Or do those tests already exist?

**Comment by brandones** [comment]
It seems like it should be possible to test any kind of Lucene search mappings using the technique described in the blog post, whether provided through annotations or programmatically.

There are no existing tests having to do with Hibernate Search / Lucene.

Yeah, actually, perhaps it would be more appropriate to write these tests for PatientName rather than some hypothetical TestObject. Since LuceneAnalyzerFactory doesn't actually have any logic it maybe doesn't make a lot of sense to test it yet. [Making it configurable](https://issues.openmrs.org/browse/TRUNK-5669) would warrant giving it some tests.

**Comment by mogoodrich** [comment]
Sounds good @brandones !

**Comment by brandones** [comment]
Actually, @mogoodrich , why don't I just write these tests against [HibernatePatientDAO](https://github.com/openmrs/openmrs-core/blob/master/api/src/main/java/org/openmrs/api/db/hibernate/HibernatePatientDAO.java)? This is where the PersonName queries [actually happen](https://github.com/openmrs/openmrs-core/blob/640cd7dd47f65c02574458f4b25d8b114aa817ed/api/src/main/java/org/openmrs/api/db/hibernate/HibernatePatientDAO.java#L194). And right now there are [no tests](https://github.com/openmrs/openmrs-core/blob/640cd7dd47f65c02574458f4b25d8b114aa817ed/api/src/test/java/org/openmrs/api/db/hibernate/HibernatePatientDAOTest.java#L28) for it.

Actually, should I be writing tests against HibernatePatientDAO directly? Or is there a way to write tests against the PatientService (or PatientDAO?), which could then run against the Hibernate DAO or any other implementation?

**Comment by mogoodrich** [comment]
There should be tests against the PatientService already, there must be a PatientServiceTest class?  Would be interesting to see if those actually cover Lucene searches for patients... (they very well might).

**Comment by brandones** [comment]
Oh whoops, you're right! The relevant tests are in PatientServiceTest. Including [coverage for Lucene stuff](https://github.com/openmrs/openmrs-core/blob/640cd7dd47f65c02574458f4b25d8b114aa817ed/api/src/test/java/org/openmrs/api/PatientServiceTest.java#L1908)! Should I close this, since it's apparently superfluous?

**Comment by mogoodrich** [comment]
Sounds good @brandones , yeah feel free to close this one.

**Pull request closed without merging**